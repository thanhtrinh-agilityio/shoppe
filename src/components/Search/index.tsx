import { ChangeEvent } from 'react'
import { Box, IconButton, Input } from '@chakra-ui/react'
import { SearchIcon } from '@chakra-ui/icons'

export interface ISearchProps {
  onChange: (e: ChangeEvent<HTMLInputElement>) => void
  onKeyDown: (e: React.KeyboardEvent<HTMLInputElement>) => void
  onClick: (e: React.MouseEvent<HTMLButtonElement>) => void
  placeholder: string
}

const Search = ({
  onChange,
  onKeyDown,
  onClick,
  placeholder,
}: ISearchProps) => {
  return (
    <Box
      borderBottom="1px"
      borderBottomColor="lightSilver"
      display="flex"
      justifyContent="space-between"
      alignItems="center"
    >
      <Input
        type="text"
        padding="0px"
        color="secondary"
        focusBorderColor="none"
        border="none"
        placeholder={placeholder}
        onChange={onChange}
        onKeyDown={onKeyDown}
      />
      <IconButton
        size="small"
        marginBottom="10px"
        type="submit"
        onClick={onClick}
        aria-label="Search database"
        icon={<SearchIcon boxSize={5} />}
      />
    </Box>
  )
}

export default Search
