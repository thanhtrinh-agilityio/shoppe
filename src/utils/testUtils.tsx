import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'

// Themes
import { customTheme } from '@themes/index'

// Contexts
import { CartProvider } from '@contexts/CartProvider'

export const customRender = (component: ReactNode) => {
  return render(
    <ChakraProvider theme={customTheme}>
      <CartProvider>{component}</CartProvider>
    </ChakraProvider>,
  )
}

export * from '@testing-library/react'
export * from '@testing-library/jest-dom'
export { customRender as render }
