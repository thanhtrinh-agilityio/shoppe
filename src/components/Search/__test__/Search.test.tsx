// Utils
import { render, fireEvent, screen } from '@utils/testUtils'

// Components
import Search from '@components/Search'

describe('Search component', () => {
  const props = {
    onClick: jest.fn(),
    onKeyDown: jest.fn(),
    onChange: jest.fn(),
    placeholder: 'Search...',
  }

  it('should render Search', () => {
    const { container } = render(<Search {...props} />)

    expect(container).toMatchSnapshot()
  })

  it('should call onChange Search ', () => {
    render(<Search {...props} />)

    const search = screen.getByRole('textbox') as HTMLInputElement

    fireEvent.change(search, {
      target: { value: 'Search Value' },
    })

    expect(props.onChange).toBeCalled()
    expect(search.value).toBe('Search Value')
  })

  it('should call onClick Icon Search ', () => {
    render(<Search {...props} />)

    const iconSearch = screen.getByRole('button')

    fireEvent.click(iconSearch)
    expect(props.onClick).toBeCalled()
  })

  it('should call onKeyDown after Enter ', () => {
    render(<Search {...props} />)

    const search = screen.getByRole('textbox')
    fireEvent.keyDown(search)

    expect(props.onKeyDown).toBeCalled()
  })
})
