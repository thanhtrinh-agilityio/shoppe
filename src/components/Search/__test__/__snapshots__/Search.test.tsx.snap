// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component should render Search 1`] = `
<div>
  <div
    class="css-zrbfnb"
  >
    <input
      class="chakra-input css-b2ycb2"
      placeholder="Search..."
      type="text"
    />
    <button
      aria-label="Search database"
      class="chakra-button css-1ljo2n9"
      type="submit"
    >
      <svg
        aria-hidden="true"
        class="chakra-icon css-bokek7"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
